include/swift/AST/DiagnosticsSema.def

```diff
@@ -5675,6 +5675,11 @@ ERROR(override_nsobject_hash_error,none,
       "'NSObject.hash(into:)' is not overridable; "
       "did you mean to override 'NSObject.hash'?", ())
 
+WARNING(synthesized_hashable_may_not_match_custom_equatable, none,
+         "automatically generated 'Hashable' implementation for type %0 "
+         "may not match the behavior of custom '==' operator", (Type))
```

Comment:
```suggestion
         "may not match behavior of custom '==' operator", (Type))
```

(Reduced written register does't use a definite article here.)

More pertinently, is there a word to describe what is meant better than "match"? To me, a simple reading would suggest that 'Hashable' should behave like '==', and of course that doesn't make sense; a reader might ignore the warning because why would they want two disparate things (a protocol and an operator) to "match"?

Reply:
"be compatible with" might be more precise, but that's still not really actionable for the user unless we spell out what "compatible" actually means. I'm not sure how verbose we want this warning to be, however. Ideally we'd be able to just stick in a reference to some relevant documentation or something.

Reply:
Maybe, “aligned”? Instead of “behavior,” maybe “semantics”? Not sure but I think we can get a little closer with some wordsmithing.

Reply:
So if we were going to really spell it out precisely, what would we say? Let's start with that and try to back off. Something like

"Any two values that compare equal with `==` must also have the same hash; because you have provided a custom `==` implementation for %0, you should also provide a custom implementation of `Hashable`."

Reply:
Yeah, you beat me to the punch. I was going to suggest something similar:

“custom ‘Equatable’ conformance without custom ‘Hashable’ conformance may cause equivalent values to have different hashes” (fixit: insert missing declarations required by ‘Hashable’)